### PR TITLE
feat: add components to /builders

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Adds all current component types to `/builders`.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/builders/__tests__/components/actionRow.test.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { APIActionRowComponent, ButtonStyle, ComponentType } from 'discord-api-types/v9';
+import { ActionRow, ButtonComponent, createComponent } from '../../src';
+
+describe('Action Row Components', () => {
+	describe('Assertion Tests', () => {
+		test('GIVEN valid components THEN do not throw', () => {
+			expect(() => new ActionRow().addComponents(new ButtonComponent())).not.toThrowError();
```
Comment (vladfrangu) [inline packages/builders/__tests__/components/actionRow.test.ts:7]: Let's add a test to ensure the output JSON from builders is valid

File: packages/builders/__tests__/components/button.test.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+			expect(() => buttonLabelValidator.parse(longStr)).toThrowError();
+		});
+
+		test('GIVEN valid style then validator does not throw', () => {
+			expect(() => buttonStyleValidator.parse(3)).not.toThrowError();
+			expect(() => buttonStyleValidator.parse(ButtonStyle.Secondary)).not.toThrowError();
+		});
+
+		test('GIVEN invalid style then validator does not throw', () => {
```
Comment (vladfrangu) [inline packages/builders/__tests__/components/button.test.ts:33]: Consistent test names please

File: packages/builders/src/components/selectMenu/SelectMenu.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	validateRequiredSelectMenuParameters,
+} from '../Assertions';
+import type { Component } from '../Component';
+import { SelectMenuOption } from './SelectMenuOption';
+
+/**
+ * Represents a select menu component
+ */
+export class SelectMenuComponent implements Component {
+	public readonly type = ComponentType.SelectMenu;
```
Comment (vladfrangu) [inline packages/builders/src/components/selectMenu/SelectMenu.ts:16]:
```suggestion
	public readonly type = ComponentType.SelectMenu as const;
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/builders/__tests__/components/actionRow.test.ts (hunk 1 above)
Comment (suneettipirneni) [reply packages/builders/__tests__/components/actionRow.test.ts:7]: Isn't that what the code for `GIVEN valid JSON input THEN valid JSON output is given` is doing?

Comment (suneettipirneni) [reply packages/builders/__tests__/components/actionRow.test.ts:7]: oh wait nvm I see

**Review by vladfrangu — APPROVED** [review]
Looks fine to me. Can you not implement something similar to application command options (relating to the toJSON() being mostly {...this}?

**Comment by suneettipirneni** [comment]
> Looks fine to me. Can you not implement something similar to application command options (relating to the toJSON() being mostly {...this}?

I can, I just didn't know if handling snake_case like this:

```
{
	...this
    snake_case_prop: this.camelCaseProp,
}
```

was preferred or not

**Comment by vladfrangu** [comment]
It's what I did for builders (and subsequently forgot to do consistently), but CC @discordjs/the-big-4 about this

**Comment by suneettipirneni** [comment]
> It's what I did for builders (and subsequently forgot to do consistently), but CC @discordjs/the-big-4 about this

Also I just realized that if there are snake_case keys, and you do `...this` you'll have the leftover camelcase key in the json. So that'll also have to be manually `delete`d off the object.

**Comment by vladfrangu** [comment]
> > It's what I did for builders (and subsequently forgot to do consistently), but CC @discordjs/the-big-4 about this
> 
> Also I just realized that if there are snake_case keys, and you do `...this` you'll have the leftover camelcase key in the json. So that'll also have to be manually `delete`d off the object.

Just replace all camelCase props on the class with snake_case 🦐

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by iCrawl** [comment]
This needs a rebase.

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
Are the changes to the yarn.lock file intentional? 👀

**Comment by suneettipirneni** [comment]
Note: CI failing because of voice tests

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**